asteroid/engine/system.py

```diff
@@ -36,7 +36,7 @@ def __init__(self, model, optimizer, loss_func, train_loader,
         self.val_loader = val_loader
         self.scheduler = scheduler
         config = {} if config is None else config
-        self.config = config
+        self.config = sanitize_configs(config)
```

Comment:
I think the config dict should not be modified. 
Instead, we replace self.none_to_string() by self.config_to_hparams(), and config2hparams contains flatten_dict, none_to_string, and the function you wrote to support tuples and lists. 
As all these functions are only related to hparams, SummaryWriter and System, we can attach it to `System`. 
Tell me if it's clear.

Reply:
I think it is. So we make an unique function for this self.none_to_string(flatten_dict(config)) where we also convert lists and tuples to Tensors

Reply:
Yes.